app.rb

```diff
@@ -4,128 +4,40 @@
   end
 
   def list_all_books
-    puts 'Database is empty! Add a book.' if @books.empty?
-    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+    @displayer.display_books(@books)
   end
 
   def list_all_persons
-    puts 'Database is empty! Add a person.' if @persons.empty?
-    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name} Age: #{person.age} Id: #{person.id}" }
+    @displayer.display_persons(@persons)
   end
 
   def create_person
-    print 'To create a student, press 1, to create a teacher, press 2 :'
-    option = gets.chomp
-
-    case option
-    when '1'
-      create_student
-    when '2'
-      create_teacher
-    else
-      puts 'Invalid input. Try again'
-    end
-  end
-
-  def create_student
-    puts 'Create a new student'
-    print 'Enter name: '
-    name = gets.chomp
-    print 'Enter student age: '
-    age = gets.chomp.to_i
-    print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase
-    case parent_permission
-    when 'n'
-      student = Student.new(age, 'classroom', name, parent_permission: false)
-      @persons << student
-      puts "Student doesn't have parent permission, can't rent a books "
-    when 'y'
-      student = Student.new(age, 'classroom', name, parent_permission: true)
-      @persons << student
-      puts 'Student created successfully'
-    end
-  end
-
-  def create_teacher
-    puts 'Create a new teacher'
-    print 'Enter name: '
-    name = gets.chomp
-    print 'Enter teacher age'
-    age = gets.chomp.to_i
-    print 'Enter teacher specialization: '
-    specialization = gets.chomp
-    teacher = Teacher.new(age, specialization, name)
-    @person << teacher
-    puts 'Teacher created successfully'
+    @persons << CreatePerson.new.create_person
   end
 
-  def create_book
-    puts 'Create a new book'
-    print 'Enter title: '
-    title = gets.chomp
-    print 'Enter an author: '
-    author = gets.chomp
-    book = Book.new(title, author)
-    @books.push(book)
-    puts "Book #{title} created successfully."
+  def create_book()
+    @books << CreateBook.new.create_book
   end
 
   def create_rental
-    puts 'Select which book you want to rent by entering its number'
-    @books.each_with_index { |book, index| puts "[#{index}] Title: #{book.title} Author: #{book.author}" }
-
-    book_id = gets.chomp.to_i
-
-    puts 'Select a person from the list by its number'
-    @persons.each_with_index do |person, index|
-      puts "[#{index}] [#{person.class.name}] Name: #{person.name} Id: #{person.id} Age: #{person.age}"
-    end
-
-    person_id = gets.chomp.to_i
-
-    print 'Date: '
-    date = gets.chomp.to_s
-
-    rental = Rental.new(date, @persons[person_id], @books[book_id])
-    @rentals << rental
-
-    puts 'Rental created successfully'
+    @rentals << CreateRental.new.create_rental(@books, @persons)
   end
 
   def list_all_rentals
-    print 'To see person rentals enter the person ID: '
```

Comment:
- [ ] Asking the user for the initial choice of options( or to quit) should have been done in the main.rb file. Please let's fix this.

Reply:
These are the deleted code